strawberry_django_plus/type.py

```diff
@@ -89,13 +91,18 @@ def _from_django_type(
 
         field = cast(StrawberryDjangoField, field)
 
-        # FIXME: Improve this...
         if not field.base_resolver:
 
-            def conn_resolver(root):
-                return getattr(root, name).all()
+            def conn_resolver(type_annotation, root, info: Info):
+                qs =  getattr(root, name).all()
+                remote_type_defs = get_args(type_annotation.annotation)
+                remote_type = eval_type(remote_type_defs[0], type_annotation.namespace, None)
+                if hasattr(remote_type, "get_queryset"):
+                    return remote_type.get_queryset(qs, info)
+                return qs
+            aware_conn_resolver = partialfunc(conn_resolver, type_annotation)
 
-            field.base_resolver = StrawberryResolver(conn_resolver)
+            field.base_resolver = StrawberryResolver(aware_conn_resolver)
```

Comment:
The `StrawberryResolver` has some methods/attributes to resolve its args. I wonder if this would make it break somehow, by either not being able to resolve the arguments, or if `type_annotation` would appear in the schema somehow.

Maybe a unit test for this case would be nice to be sure that that won't happen.

Reply:
Renamed  type_annotation as a preventive measure. However, tests for Python v3.8/3.9 are failing due to a "broken" partial in functools. Added a workaround. I'll see if I can create a test case for the `conn_resolver` function